### PR TITLE
Fix gitlab commit sha

### LINKED ISSUE #5486: [Bug][gitlab] Not fetching correctly squashed commits from merge requests
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

The fetching of the merge request information from GitLab APIs is not completely correct because the option "squash on merge" of GitLab is not managed correctly.
To be more specific when a MR is squashed the related commit sha is not reported inside the field "merge_commit_sha" that we are fetching from GitLab APIs but is instead inside the "squash_commit_sha" field. The information about the squashing option enabled or disabled is inside the "squash" field.

Gitlab docs: https://docs.gitlab.com/ee/api/merge_requests.html#response

### What do you expect to happen

I would have expected to have the "merge_commit_sha" field in the "pull_requests" table be populated to proceed with the DORA metrics validation.

### How to reproduce

- Have a working version of DevLake
- Integrate it with a GitLab project
- In the integrated Gitlab project set as "Required" the squash option (under Settings -> Merge Request ->  Squash commits when merging)
- Create a Merge request in the project and merge it 

### Anything else

_No response_

### Version

v0.17.0

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Marco-De-Stefani**
### Summary
Fix the fetching of squashed and fast-forwarded commits shas from GitLab

### Does this close any open issues?
Closes #5486

### CODE REVIEW

**Comment by klesh** [comment]
Looks good to me. Can you attach some screenshots to prove that the code is working as expected? Thanks.

**Comment by Marco-De-Stefani** [comment]
Yes, sure! 
GitLab API call: 
Screenshot 2023-06-16 alle 14 24 43

Query in DevLake DB
image

File: backend/plugins/gitlab/tasks/mr_extractor.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ type MergeRequestRes struct {
 	MergedAt        *api.Iso8601Time `json:"merged_at"`
 	ClosedAt        *api.Iso8601Time `json:"closed_at"`
 	MergeCommitSha  string           `json:"merge_commit_sha"`
+	SquashCommitSha string           `json:"squash_commit_sha"`
+	DiffHeadSha     string           `json:"diff_head_sha"`
```
Comment (klesh) [inline backend/plugins/gitlab/tasks/mr_extractor.go:55]: @Marco-De-Stefani Thanks. I checked the Gitlab REST doc, looks like the field `diff_head_sha` doesn't exist. 
According to the `sha` field description from https://docs.gitlab.com/ee/api/merge_requests.html#response, I think it is the correct one we need.

**Review by klesh — COMMENTED** [review]
The field `diff_head_sha` doesn't seem right, can you double check the comment I left? Thanks

File: backend/plugins/gitlab/tasks/mr_extractor.go (hunk 1 above)
Comment (Marco-De-Stefani) [reply backend/plugins/gitlab/tasks/mr_extractor.go]: Fixed.
I have a question though, from my API tests it seems that that field is always populated (the other 2 are populated only when the MR is merged, this one is always there and gets updated), could that cause some issues down the line?

**Comment by klesh** [comment]
@Marco-De-Stefani The e2e-test has failed. Do you think you could fix it?
Maybe we should extract those 3 fields first and do the fallback in the `mr_convertor.go`

**Comment by Marco-De-Stefani** [comment]
I'll try to do that after some work stuff, thanks a lot @klesh

**Comment by Marco-De-Stefani** [comment]
@klesh it should be fine now, can you re-trigger the workflow?

**Comment by klesh** [comment]
@Marco-De-Stefani Hi, Seems like there are multiple places that need to be updated.

I suggest that you delete those CSVs and rerun the `e2e` test, it will create new ones, then we just need to verify if they are expected

1. Try the following way to run the e2e test case locally and identify the failed CSVs, they are likely located inside the ` snapshot_tables` folder:
```sh
cd backend
E2E_DB_URL="mysql://merico:merico@localhost:3306/lake_e2e?charset=utf8mb4&parseTime=True"
go test github.com/apache/incubator-devlake/plugins/gitlab/e2e
```
2. identify the failed CSV, delete them
3. rerun the e2e test
4. check the newly generated CSV

Feel free to reach out to me if any problem, your contribution is solid, the `e2e` tests are to make sure changes won't be overwritten accidentally.

**Review by mappjzc — APPROVED** [review]
The e2e error will be fixed by #5566

File: backend/plugins/gitlab/models/mr.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ type GitlabMergeRequest struct {
 	SourceBranch     string `gorm:"type:varchar(255)"`
 	TargetBranch     string `gorm:"type:varchar(255)"`
 	MergeCommitSha   string `gorm:"type:varchar(255)"`
+	SquashCommitSha  string `gorm:"type:varchar(255)"`
```
Comment (keon94) [inline backend/plugins/gitlab/models/mr.go:42]: The new fields need DB migrations

Comment (keon94) [reply backend/plugins/gitlab/models/mr.go:42]: image
cc @mappjzc

Comment (keon94) [reply backend/plugins/gitlab/models/mr.go:42]: Ah, https://github.com/apache/incubator-devlake/pull/5566 is doing it. Nvm then.

**Pull request merged**